src/services/expensesService.js

```diff
@@ -0,0 +1,69 @@
+const { data } = require('../data');
+
+const getAllExpenses = () => {
+  return data.expenses;
+};
+
+const getById = (id) => {
+  return data.expenses.find((exp) => exp.id === parseInt(id));
+};
+
+const normalizeCategory = (categories) => {
+  return Array.isArray(categories) || !categories ? categories : [categories];
+};
+
+const filterExpensesById = (id) => {
+  data.expenses = data.expenses.filter((exp) => exp.userId === parseInt(id));
```

Comment:
Code quality issue: This function `filterExpensesById` alters the original `data.expenses` array. It's a better practice to avoid mutating the original data. Instead, you can return a new array with the filtered results.